src/CSV.jl

```diff
@@ -20,35 +20,40 @@ end
 
 Base.showerror(io::IO, e::Error) = println(io, e.msg)
 
+struct Column{T, P} <: AbstractVector{T}
+    tape::Vector{UInt64}
+    len::Int
+    e::UInt8
+    catg::Bool
+    refs::Union{Vector{String}, Nothing}
+    buf::Vector{UInt8}
+    sentinel::UInt64
+end
+
+_eltype(::Type{T}) where {T} = T
+_eltype(::Type{PooledString}) = String
+_eltype(::Type{Union{PooledString, Missing}}) = Union{String, Missing}
+
+Base.size(c::Column) = (c.len,)
+Base.IndexStyle(::Type{<:Column}) = Base.IndexLinear()
+
 struct File
     name::String
     names::Vector{Symbol}
     types::Vector{Type}
     rows::Int64
     cols::Int64
-    e::UInt8
-    categorical::Bool
-    refs::Vector{Vector{String}}
-    buf::Vector{UInt8}
-    tapes::Vector{Vector{UInt64}}
+    columns::Vector{AbstractVector}
```

Comment:
The change here is moving from storing the raw tapes + additional info to just storing the final, `Column` structures directly. That moves a bit of work from `getproperty` on `CSV.File` previously to `CSV.File` construction.

Reply:
The main advantage of this change is it removes a bunch of duplicated logic that existed between tables.jl and iteration.jl before, since row iterating on `CSV.File` now just makes a lazy `Row` wrapper that indexes the already-created `CSV.Column`s.